src/App.js

```diff
@@ -160,6 +165,39 @@ const DashboardStack = () => {
  */
 const CameraPermissionScreen = () => null;
 
+/**
+ * Stack of screens dedicated to the token swap process
+ */
+const SwapStack = ({ navigation }) => {
+  const Stack = createStackNavigator();
+  const [initialRoute, setInitialRoute] = useState('TokenSwapLoadingScreen');
+  /** @type {null|'loading'|'loaded'|'error'} */
+  const allowedTokensStatus = useSelector((state) => state.tokenSwap.loadAllowedTokensStatus);
+
+  useEffect(() => {
+    // When we load the tokens, move to the token swap main screen
+    if (allowedTokensStatus === 'loaded') {
+      setInitialRoute('TokenSwap');
+      navigation.replace('TokenSwap');
+    }
```

Comment:
I'm not fully against this but we usually execute the loading on the screen itself, so we can control the components loading/error/success on the same place depending on the redux state.

What do you think about it?

Reply:
I was copying the behavior of the `SendStack` but I also think that the replace is not doing much here.